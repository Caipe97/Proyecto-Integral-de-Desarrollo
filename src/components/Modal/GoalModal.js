import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button, ProgressBar } from 'react-bootstrap';

const GoalModal =(props) =>{
  const [modalShow, setModalShow] = useState(false);

  const calculateCurrentCaloriesOfGoal = goal => {
    let currentCaloriesPerObjective = 0;
    for (let i = 0; i < goal.objectives.length; i++) {
      const objective = goal.objectives[i];
      currentCaloriesPerObjective += objective.currentCalories;
    }
    return currentCaloriesPerObjective;
  }

  const isGoalEditable = goal => {
    let dateStart = new Date(goal.dateStart);
    dateStart = new Date(dateStart.setHours(dateStart.getHours() + 3));
    const today = new Date();
    if(dateStart.getFullYear() > today.getFullYear()){
      return true;
    } else if(dateStart.getFullYear() === today.getFullYear()){
      if(dateStart.getMonth() > today.getMonth()){
        return true;
      } else {
        return false;
      }
    } else {
      return false;
    }
  }

  return (
    <>
      <Button  type="button" variant="primary" style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}} onClick={() => setModalShow(true)}>
        Mas info
      </Button>
      
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {props.goal.name} 
          </Modal.Title>
          <Button type="button" onClick={() => setModalShow(false)} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>

        </Modal.Header>
        <Modal.Body>
          <h4>{props.goal.dateStart.toString().replace('T',' ').substring(0,7)}</h4>
          <p>
            Objetivos:
          </p>
          
          {props.goal.objectives.map(objective => {
            return(
              <div key={objective.foodCategoryId} style={{textAlign:'center',top:'3px',background:'#B6E052', margin: "5px", padding: 5, borderRadius: 10}}>
                <p style={{margin: 0}}>{objective.foodCategory.name}: {objective.currentCalories} de {objective.objectiveCalories}</p>
                <ProgressBar animated 
                  variant={(objective.currentCalories/objective.objectiveCalories)*100 > 65 ? "danger" : "success"}  
                  now={(objective.currentCalories/objective.objectiveCalories)*100} 
                  label={`${(objective.currentCalories/objective.objectiveCalories)*100}%`}
                />
                  
              </div>
            );
            })
          }
          {
          props.goal.totalCalories > 0 
          ?  <>
              <p>Calorias totales consumidas: {calculateCurrentCaloriesOfGoal(props.goal)}/{props.goal.totalCalories}:</p> 
              <ProgressBar animated 
                variant={(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100 > 65 ? "danger" : "success"}  
                now={(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100} 
                label={`${(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100}%`}
              />
            </>
          : null
          }
        </Modal.Body>
        <Modal.Footer>
          { isGoalEditable(props.goal)
            ? <Button type="button" onClick={() =>  {props.onUpdateCurrentGoalInState(props.goal); props.history.push("/goals", {goal: props.goal})}} style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>Editar</Button>
            : null
          }
          <Button type="button" onClick={() =>  props.onDeleteGoal(props.goal.goalId, props.goal.userId)} style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>Eliminar</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
  
  export default GoalModal;