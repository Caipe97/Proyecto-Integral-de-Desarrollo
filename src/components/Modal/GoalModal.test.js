import React from 'react';
import { shallow } from 'enzyme'; 
import GoalModal from './GoalModal';
//    <GoalModal goal={row} history={this.props.history} onDeleteGoal={this.props.onDeleteGoal} onUpdateCurrentGoalInState={this.props.onUpdateCurrentGoalInState} />
          //props.goal.objectives.map
describe('GoalModal tests', () => {
    const historyMock = { push: jest.fn() };
    const preventDefault = { preventDefault: jest.fn() };
    const mockPropsSignIn = {
        goal: {name:'enero',dateStart:"2021-09-30T14:09:34.756Z",totalCalories:300,objectives:[]}, 
        onDeleteGoal: jest.fn(),
        onUpdateCurrentGoalInState:jest.fn(),
        preventDefault:jest.fn()
    };
    let wrapper;
    

    beforeEach(() => {
        wrapper = shallow(<GoalModal {...mockPropsSignIn} history={historyMock}/>);
     
    })

    it('expect to render GoalModal component to render', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('expect to change  onclick buttons for show more info',  () => {
         wrapper.find('[type="button"]').at(0).simulate('click');
    })
    it('expect to change  onclick buttons for show more info for hide this',  () => {
        wrapper.find('[type="button"]').at(1).simulate('click');
   })
   it('expect to change  onclick buttons for delete goal',  () => {
    wrapper.find('[type="button"]').at(2).simulate('click',preventDefault);
    //expect(wrapper.instance().props.onDeleteGoal).toHaveBeenCalledTimes(1);
})
})
