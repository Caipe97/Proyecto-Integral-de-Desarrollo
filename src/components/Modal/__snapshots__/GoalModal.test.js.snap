// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalModal tests expect to render GoalModal component to render 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(18, 207, 90)",
        "borderColor": "rgb(18, 207, 90)",
      }
    }
    type="button"
    variant="primary"
  >
    Mas info
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    restoreFocus={true}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        enero
      </ModalTitle>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "color": "black",
          }
        }
        type="button"
        variant="primary"
      >
        X
      </Button>
    </ModalHeader>
    <ModalBody>
      <h4>
        2021-09
      </h4>
      <p>
        Objetivos:
      </p>
      <div
        key="1"
        style={
          Object {
            "background": "#B6E052",
            "borderRadius": 10,
            "margin": "5px",
            "padding": 5,
            "textAlign": "center",
            "top": "3px",
          }
        }
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          fideos
          : 
          3000
           de 
          3000
        </p>
        <ProgressBar
          animated={true}
          isChild={false}
          label="100.00%"
          max={100}
          min={0}
          now={100}
          srOnly={false}
          striped={false}
          variant="danger"
        />
      </div>
      <div
        key="2"
        style={
          Object {
            "background": "#B6E052",
            "borderRadius": 10,
            "margin": "5px",
            "padding": 5,
            "textAlign": "center",
            "top": "3px",
          }
        }
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          carnes
          : 
          3000
           de 
          2000
        </p>
        <ProgressBar
          animated={true}
          isChild={false}
          label="150.00%"
          max={100}
          min={0}
          now={150}
          srOnly={false}
          striped={false}
          variant="danger"
        />
      </div>
      <p>
        Calorias totales consumidas: 
        /
        300
        :
      </p>
      <ProgressBar
        animated={true}
        isChild={false}
        label="NaN%"
        max={100}
        min={0}
        now={NaN}
        srOnly={false}
        striped={false}
        variant="success"
      />
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(18, 207, 90)",
            "borderColor": "rgb(18, 207, 90)",
          }
        }
        type="button"
        variant="primary"
      >
        Eliminar
      </Button>
    </ModalFooter>
  </Modal>
</Fragment>
`;
